### ISSUE TITLE
Unreachable code in the next line when using `read_excel` function of `polars` module

### ISSUE THREAD

**Issue by arnabanimesh** [pylint-dev/pylint#10016]
### Bug description
```python
# pylint: disable=missing-module-docstring

import polars as pl

df = pl.read_excel("data.xlsx")
print(df.columns)
```

### Command used

`pylint hi.py`

### Pylint output
```
************* Module hi
hi.py:6:0: W0101: Unreachable code (unreachable)

------------------------------------------------------------------
Your code has been rated at 6.67/10 (previous run: 0.00/10, +6.67)
```

### Expected behavior

No warnings

### Pylint version
```
pylint 3.3.1
astroid 3.3.5
Python 3.12.5 (main, Aug 14 2024, 04:23:18) [MSC v.1929 64 bit (AMD64)]
```
### OS / Environment

Powershell in Windows 11 23H2

### Additional dependencies
```
polars==1.9.0
fastexcel==0.11.6
```

**Comment by Pierre-Sassoulas** [comment]
Fixed in https://github.com/pylint-dev/pylint/commit/4f6c2412097120587d6fc01fa0d344c49780f225

**Comment by Nicolas-Bechstedt** [comment]
@Pierre-Sassoulas As of pylint 4.0.6 and polars 1.44.2 (latest), this is still not fixed.
Should this issue be reopened or a new issue be created?

pylint --version:
```
pylint 4.0.6
astroid 4.0.3
Python 3.14.7 | packaged by Anaconda, Inc. | (main, Aug 14 2026, 10:12:46) [MSC v.1942 64 bit (AMD64)]
```

test.py:
```python
"""docstring"""

import polars as pl


def test() -> None:
    """docstring"""
    pl.read_excel("abc.xlsx")
    print()
```

pylint test.py:
```
************* Module test
test.py:9:4: W0101: Unreachable code (unreachable)

------------------------------------------------------------------
Your code has been rated at 7.50/10 (previous run: 7.50/10, +0.00)
```

**Comment by Pierre-Sassoulas** [comment]
Yes please, thank you for the reproducer (didn't test yet)

**Comment by Nicolas-Bechstedt** [comment]
Also confirmed with the latest version, since I had the default conda version, but it also still exists in pylint 4.0.8

**Comment by Nicolas-Bechstedt** [comment]
But since I cannot reopen the issue I would go ahead and make a new one then 👍
